dist/tools/coccinelle/check.sh

```diff
@@ -59,6 +59,7 @@ coccinelle_checkone() {
                 echo "$OUT" | {
                     # see https://stackoverflow.com/a/30064493/11921757 for why we
                     # use a sub shell here
+                    OLD_IFS="${IFS}"
```

Comment:
Can you add a comment on this? Is not clear right away for me the reason.

Reply:
I did comment on that below. Or do you mean I should add a comment why I store the old `IFS`?

Reply:
Yes oh why you stored the `OLD_IF`, why not restore them when calling `_annotate_diff` for example?

Reply:
How should I restore them, if I do not remember them first?

Reply:
But fine, will comment something like that if that is not obvious :-)

Reply:
Done

Reply:
> Yes oh why you stored the OLD_IF, why not restore them when calling _annotate_diff for example?

With this I mean : `IFS="${OLD_IFS}" _annotate_diff "$DIFFFILE" "$DIFFLINE" "$DIFF"`, or Am I missing something and making stupid comments?

Reply:
> IFS="${OLD_IFS}" _annotate_diff "$DIFFFILE" "$DIFFLINE" "$DIFF"

Instead of doing it twice in the body on `_annotate_diff`

Reply:
Ah you mean like this https://github.com/RIOT-OS/RIOT/pull/15795/commits/a2d7eb9fde8f1be8c95810e39a9166f874eeb8f3? Let's give that a try. I will adopt the other scripts if it works... but this is not about a comment anymore ;-), so I am not 100% sure this is what you meant.

Reply:
> But fine, will comment something like that if that is not obvious :-)

Well It would have been clearer to me if the comment had been something like `keep leading and trailing spaces when passing the diff but restore them before calling annotations for reason??` why are we restoring them? I did not see the result before the fixups so I don't know what the result was), so yes maybe evident for you but I jumped in late, and I don't know how those  `github_annotate_%` functions are implemented unless I go look at the code and try to figure out the outcome without handling new lines.

Reply:
Of course, restoring to default makes sense in its self, so such a verbose comment is not necessarily needed.

Reply:
[Here is an example](https://github.com/RIOT-OS/RIOT/pull/16076/files#diff-24352765bf21d083ab5a71dae2b6f18cceb2c8f5bf9157b7fa4a23b4f7d0f53cR171) of how the annotations look like without this PR:

```diff
--- a/core/include/xfa.h
+++ b/core/include/xfa.h
@@ -171,7 +171,7 @@ _Pragma("GCC diagnostic ignored "-Warray-bounds"")
*/
#define XFA_ADD_PTR(xfa_name, prio, name, entry)      _XFA_CONST(xfa_name, 5_ ## prio) -    const __typeof__(entry) xfa_name ## _ ## prio ## _ ## name = entry
+    const __typeof__(entry)xfa_name ## _ ## prio ## _ ## name = entry

/**
* @brief Calculate number of entries in cross-file array
```

With this PR it would look like how it is supposed to:

```diff
--- a/core/include/xfa.h
+++ b/core/include/xfa.h
@@ -171,7 +171,7 @@ _Pragma("GCC diagnostic ignored "-Warray-bounds"")
*/
#define XFA_ADD_PTR(xfa_name, prio, name, entry)      _XFA_CONST(xfa_name, 5_ ## prio) \
-    const __typeof__(entry) xfa_name ## _ ## prio ## _ ## name = entry
+    const __typeof__(entry)xfa_name ## _ ## prio ## _ ## name = entry

/**
* @brief Calculate number of entries in cross-file array
```

(we shouldn't discuss the correctness of that false positive though ;-)). So for this to work, we need the lines to be read correctly, e.g. with all the backslashes `\` (that's why the `-r` is there), and with all the surrounding whitespaces (which is why we set `IFS` to `''` as this way white spaces are not read as separators). However, since the `github_annotate_*` commands need the normal `IFS` settings to work, we need to store the old value of `IFS` and restore it when calling those functions.

Reply:
> Ah you mean like this [a2d7eb9](https://github.com/RIOT-OS/RIOT/commit/a2d7eb9fde8f1be8c95810e39a9166f874eeb8f3)? Let's give that a try. I will adopt the other scripts if it works... but this is not about a comment anymore ;-), so I am not 100% sure this is what you meant.

Nope, coccinelle now does not generate the correct annotation more. See [this output](https://github.com/RIOT-OS/RIOT/pull/15795/checks?check_run_id=1962257292#step:5:20).

Reply:
Will revert a2d7eb9

Reply:
OK!